lib/superagent-mock.js

```diff
@@ -142,7 +142,14 @@ function mock (superagent, config, logger) {
       }
     }
 
-    fn(error, response);
+    if (typeof context.delay === 'number') {
+      setTimeout(function () {
+        fn(error, response);
```

Comment:
Be careful here because response is a global variable in the whole mock function. Thus if you do two requests in a row with delays and the second request sends a response before the first, then response of the first request will be the response of the second request actually. Would be great if you add this scenario in your test 👍